frontend/src/metabase/questions/components/CollectionBadge.tsx

```diff
@@ -41,13 +51,18 @@ function CollectionBadge({ className, collection, isSingleLine }) {
       {collection.getName()}
     </Badge>
   );
-}
-
-CollectionBadge.propTypes = propTypes;
+};
 
-export default Collection.load({
-  id: (state, props) => props.collectionId || "root",
+export const CollectionBadge = Collection.load({
+  id: (state: State, props: { collectionId?: CollectionId }) =>
+    props.collectionId || "root",
   wrapped: true,
   loadingAndErrorWrapper: false,
   properties: ["name", "authority_level"],
-})(CollectionBadge);
+})(CollectionBadgeInner) as ComponentType<
```

Comment:
We have to account for the fact that `Collection.load` takes in a `collectionId` and returns `collection` for the `CollectionBadgeInner`, which is why the type is the way it is